src/sage/rings/polynomial/multi_polynomial_ideal.py

```diff
@@ -1046,38 +1048,38 @@ def triangular_decomposition(self, algorithm=None, singular=singular_default):
 
         EXAMPLES::
 
-            sage: P.<e,d,c,b,a> = PolynomialRing(QQ, 5, order='lex'); P.rename("P")
+            sage: P.<e,d,c,b,a> = PolynomialRing(QQ, 5, order='lex')
             sage: I = sage.rings.ideal.Cyclic(P)
```

Comment:
Why did you remove `P.rename("P")`? It shows off an idiom that may be useful.

Reply:
It actually gives an error for instances of this class.

This was masked by this whole file not being doctested (because of a mistake how skipping files was implemented).

Reply:
OK. But then I wonder if there was a regression but no one complained.

Reply:
I asked the question in sage-devel.

Reply:
Note that I introduced the version with `P.rename("P")` not very long ago. 
Probably I tested it only in a modularized distribution that replaces the Singular-based implementation by the polydict implementation.

Reply:
OK. I now understand the situation.

I am OK with the changes.